pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -477,13 +516,47 @@ stdenv.mkDerivation ({
     # TODO: fetch the self from the fixpoint instead
     haddockDir = self: if doHaddock then "${docdir self.doc}/html" else null;
 
-    env = shellFor {
-      packages = p: [ drv ];
-      inherit shellHook;
-    };
-
+    envFunc = { withHoogle ? false, ... } @ args:
+      let
+        name = "ghc-shell-for-${drv.name}";
+
+        withPackages = if withHoogle then ghcWithHoogle else ghcWithPackages;
+
+        ghcEnvForBuild = buildHaskellPackages.ghcWithPackages (_: setupHaskellDepends);
```

Comment:
Could you add a comment about why `ghcWithPackages` is being pulled from `buildHaskellPackages`?

Reply:
Comment has been added